index.js

```diff
@@ -51,87 +50,86 @@ function RFID (hardware, options, callback) {
   self.hardware = hardware;
   self.ready = false;
   self.irq = hardware.digital[2];
-  self.irqcallback = function () {
-    self.emit('irq', null, 0);
-  };
-  self.irq.on('fall', self.irqcallback);
+  self.irq.input();
+
   self.nRST = hardware.digital[1];
 
-  self.nRST.output();
-  self.nRST.low(); // Toggle reset every time we initialize
+  // Toggle reset every time we initialize
+  self.nRST.write(false, function continueInit() {
 
-  self.i2c = new hardware.I2C(PN532_I2C_ADDRESS);
+    self.i2c = new hardware.I2C(PN532_I2C_ADDRESS);
 
-  // This function only exists on the Tessel 1
-  if (typeof self.i2c._initialize === 'function') {
-    self.i2c._initialize();
-  }
+    // This function only exists on the Tessel 1
+    if (typeof self.i2c._initialize === 'function') {
+      self.i2c._initialize();
+    }
 
-  self.numListeners = 0;
-  self.listening = false;
+    self.numListeners = 0;
+    self.listening = false;
 
-  self.autoReset = true;
-  self.resetTimeout = 500;
-  if (options) {
-    if (options.hasOwnProperty('listen')) {
-      self.autoReset = options.listen;
-    }
-    if (options.hasOwnProperty('delay')){
-      self.resetTimeout = options.delay;
+    self.autoReset = true;
+    self.resetTimeout = 300;
+    if (options) {
+      if (options.hasOwnProperty('listen')) {
+        self.autoReset = options.listen;
+      }
+      if (options.hasOwnProperty('delay')){
+        self.resetTimeout = options.delay;
+      }
     }
-  }
 
-  self.irq.input();
-  setTimeout(function () {
-    self.nRST.high();
-    self._getFirmwareVersion(function (err, version) {
-      if (!version) {
-        self.emit('error', err);
-        if (callback) {
-          callback(err);
+    setTimeout(function () {
+      self.nRST.write(true, function continueAfterRst() {
```

Comment:
Implemented a callback here as well. At this point, the reset line has been pulled back up and we can be sure the device has finished resetting.